LEAF_Request_Portal/sources/Form.php

```diff
@@ -1678,13 +1704,21 @@ public function hasWriteAccess($recordID, $categoryID = 0, $indicatorID = 0)
         // find out if explicit permissions have been granted to any groups
         if (count($multipleCategories) <= 1)
         {
-            $vars = array(':categoryID' => \Leaf\XSSHelpers::xscrub($categoryID),
-                          ':userID' => $this->login->getUserID(), );
-            $resCategoryPrivs = $this->db->prepared_query('SELECT * FROM category_privs
-                                                        LEFT JOIN users USING (groupID)
-                                                        WHERE categoryID=:categoryID
-                                                            AND userID=:userID
-            												AND writable=1', $vars);
+            $resCategoryPrivs = null;
+            $cacheHash = 'hasWriteAccess_catPrivs_'.$categoryID.$this->login->getUserID();
+            if(isset($this->cache[$cacheHash])) {
+                $resCategoryPrivs = $this->cache[$cacheHash];
+            }
+            else {
+                $vars = array(':categoryID' => $categoryID,
+                              ':userID' => $this->login->getUserID());
+                $resCategoryPrivs = $this->db->prepared_query('SELECT * FROM category_privs
```

Comment:
Does this need to be a "select *"?

Reply:
Looks like it'd be safe to change this to a COUNT(*)